appGeo/src/app/game/game.component.ts

```diff
@@ -60,30 +63,201 @@ export class Game implements OnInit {
+      }
+
+      //set all other players to civilians
+      for(let i = 0; i < playerArr.length; i++) {
+        let player = playerArr[i]
+        let civilian = new Civilian()
+        civilian.init(player.getUserID(), player.getUsername(), player.getLocation(), player.getAliveStatus())
+
+        roledPlayers.set(civilian.getUserID, civilian)
+      }
+
+      //replace unroled players with roles
+      this.#setPlayers(roledPlayers)
+
+      this.startTime = new Date();
+
+      if(this.gameRules.isScheduledEnd) {
+        this.endTime = new Date(this.startTime.getTime() + 
+                                (this.gameRules.getGameLengthHours() * 60 * 60 * 1800))
+        const endJob = scheduleJob(this.getEndTime(), function() {this.#endProcess()})
+        this.#scheduledJobs.set(END_JK, endJob)
+      }
+
+      //TODO: this is where the 3 timers for voting will be set
+
+      this.#setGameActive(ACTIVE)
+
+      return SUCCESS
+
+  }
+
+  /* All checks to be run before the game starts 
+  */
+  preGameChecks() {
+      if(this.getPlayerCount() < this.gameRules.getMinPlayers()) {
+          return FAILURE
+      } else if (this.getGameActive() == ACTIVE) {
+          return FAILURE
+      } else if(this.getGameScheduled() == SCHEDULED) {
+          return FAILURE
+      }
+
+      return SUCCESS
+  }
+
   startGame() {
-    this.#setGameActive(ACTIVE)
-    return SUCCESS;
+    if(this.preGameChecks() == FAILURE) {
+      return FAILURE
+    }
+
+    return this.#startProcess();
 
     //Randomly generate killer objects for 20% of the players and civilians for the rest
     //Rebuild the players list with these objects
     //TODO
   }
 
-  endGame() {
-      this.#setGameActive(INACTIVE)
+  scheduleStart(date: Date) {
+
+      if(this.preGameChecks() == FAILURE) {
+          return FAILURE
+      }
+
+      const job = scheduleJob(date, function() {this.#startProcess()});
+      this.#scheduledJobs.set(START_JK, job) //adds job to list of jobs running
+
+      this.#setGameScheduled(SCHEDULED)
       return SUCCESS;
+  }
+
+  cancelScheduledStart() {
+      this.#scheduledJobs.get(START_JK).cancel()
+      this.#setGameScheduled(UNSCHEDULED)
+      return SUCCESS;
+  }
+
+  winningCondition(){
+    var killers_left = this.killersRemaining();
+    var civilians_left = this.civiliansRemaining();
+
+    var min_civ; //minimum number of civilians required to prevent killer win
+
+    if(this.gameRules.isWipeoutEnd()) {
+      min_civ = 1;
+    } else {
+      min_civ = killers_left;
+    }
+
+    if (killers_left > 0 && !(civilians_left >= min_civ)){
+      //Killers have won!
+      return KILLER;
+    } else if (killers_left == 0 && civilians_left > 0){
+      //Civilians have won!
+      return CIVILIAN;
+    } else if (this.getGameActive() == INACTIVE && civilians_left > 0) {
+      //Civilians have won a time victory or a premature end has occured
+      return CIVILIAN;
+    }
+    //we do not currently handle if both are wiped out
```

Comment:
Yeah, that's why I didn't add a case for it. But I figured putting a comment doesn't hurt in case a bug comes up related to that later (such as somehow the game starts with no players or something)